packages/react-primitives/README.md

```diff
@@ -3,28 +3,78 @@
+
+```
+"@slashid/react-primitives": "workspace:*"
+```
+
+## Build
+
+This library uses [vanilla-extract](https://vanilla-extract.style/) to implement type safe stylesheets. There is no build process involved, as the consuming app will import the source files and then build the styles accordingly. Check the [vanilla-extract integration docs](https://vanilla-extract.style/documentation/integrations/vite/) for more information.
+
+## Components
+
+The public interface is defined by the exports in the `src/main.ts` file. Consuming apps can import them from the package root - `import * from '@slashid/react-primitives'`.
+Below we outline the common steps to use the components.
 
 ### Theme
 
-Imported directly
+The first thing to do is to wrap the consuming app with the `<ThemeRoot>` component. This will set up the CSS variables necessary for the components to work.
+
+### Text customisation & translations
+
+We expose text related components and utilities with two goals in mind:
+
+- having a consistent text style across the app
+- making it easy to replace the built-in text with custom strings, enabling i18n
+
+In order to do that, a couple of steps are required as outlined in the following sections.
+
+#### Text config
+
+Text config is a record with string keys pointing to string values. These are app specific, so the library does not come with any strings that are built in. Instead, the consuming app is responsible for providing the configuration in that shape:
+
+```ts
+export const textConfig = {
+  "text.hello": "Hello",
+  "text.world": "World",
+};
+
+export type TextConfig = typeof textConfig;
+```
+
+#### Wrapping the app with the `<TextProvider>`
+
+The text config we created in the previous section needs to be passed to the `<TextProvider>` that acts as a wrapper for the app, making the config available to the text components.
+
+```jsx
+import { TextProvider } from "@slashid/react-primitives";
+import { textConfig } from "./text";
 
-### Public interface
+export function App({ children }) {
+  return <TextProvider text={textConfig}>{children}</TextProvider>;
+}
+```
 
-Use exports to define what can be imported from the top level VS a direct import from a nested path.
+#### Using the `<Text>` component
 
-### Build
+Since the library does not know about the text config specific to the app, we need to specify the type of the text config to be used to get type safety. This is done by creating a type alias for the text config and creating a wrapper for the `<Text>` component that uses it:
 
-Consuming app is responsible for building the bundle.
+```jsx
+import { Text as BaseText, TextProps } from "@slashid/react-primitives";
+import { TextConfig } from "./text";
 
-### Text
+type Props = TextProps<TextConfig>;
 
-Text primitives are implemented here, translations are implemented in the consuming app.
-Wrap the text primitives in a `Text` component that will handle the translation.
+export const Text: React.FC<Props> = (props) => {
```

Comment:
Replace this with a factory function like
```
const { Text, TextProvider } = createText(config)
```

Reply:
@kaspermroz @jake-slashid after a couple of iterations I decided to leave this as it is following this reasoning:
- `textConfig` is a value, not a type
- the keys must remain the same, but the values in this object can/are likely to change (eg in the jump page we have i18n)

So passing the `textConfig` to the factory function implies you cannot change these values during runtime, which is incorrect.

The only drawback of the current approach is having to wrap around the base `<Text>` component, which IMO is acceptable given the fact that this is still internal and the change was already made for the `@slashid/react` package.